File: backend/pokerogue/src/main/java/com/pokerogue/helper/ability/dto/AbilityPokemonResponse.java
package com.pokerogue.helper.ability.dto;

import com.pokerogue.helper.pokemon.data.Pokemon;
import java.util.List;

public record AbilityPokemonResponse(
        String id,
        Long pokedexNumber,
        String koName,
        String image,
        List<AbilityTypeResponse> pokemonTypeResponses
) {
    public static AbilityPokemonResponse of(
            Pokemon pokemon,
            String image,
            List<AbilityTypeResponse> pokemonTypeResponses
    ) {
        return new AbilityPokemonResponse(
                pokemon.getId(),
                (long) pokemon.getPokedexNumber(),
                pokemon.getName(),
                image,
                pokemonTypeResponses
        );
    }
}

File: android/remote/src/main/java/poke/rogue/helper/remote/dto/response/ability/AbilityPokemonResponse.kt
package poke.rogue.helper.remote.dto.response.ability

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import poke.rogue.helper.remote.dto.response.type.PokemonTypeResponse

@Serializable
data class AbilityPokemonResponse(
    val id: String,
    val pokedexNumber: Long,
    @SerialName("koName")
    val name: String,
    val image: String,
    val pokemonTypeResponses: List<PokemonTypeResponse>,
)
